cluster/charts/crossplane-types/crds/apiextensions.crossplane.io_compositions.yaml

```diff
@@ -171,8 +172,8 @@ spec:
                 provisioned using this composition will be created.
               type: string
           required:
-          - from
-          - to
+          - compositeTypeRef
+          - resources
           - writeConnectionSecretsToNamespace
```

Comment:
Not directly related but I think we can make `writeConnectionSecretsToNamespace` optional since there are compositions without any connection secret being published like networks.

Reply:
Good point 👍 I think someone mentioned that in slack recently

Reply:
@muvaf I updated this to optional, but wasn't sure how we want to handle when it is not supplied. I opted to keep it as a `string` instead of `*string`, which means that we will just pass along a `""` to the managed resource if it is not set explicitly. In that case, we are basically deferring the error handling of passing an empty namespace to the managed resource controller. What do you think about that?

Reply:
I believe this is used only for choosing the namespace of connection secret of the composite resource [here](https://github.com/crossplane/crossplane/blob/master/pkg/controller/apiextensions/composite/api.go#L275). The conn secret ref of managed resources are left to the user to configure either in the template or via patches.

So, if you'd like to tackle that in this PR, I'd suggest making it `*string` since it's optional and put a check in that [APIConfigurator](https://github.com/crossplane/crossplane/blob/master/pkg/controller/apiextensions/composite/api.go#L270) that will return early if that field is not given. I can also tackle it in another PR if you'd like to.

Reply:
@muvaf yep that's what I was looking at. So to be clear, we would just skip setting `WriteConnectionSecretToReference` instead of setting it with a `""` namespace. The other option would be returning an error, but that would obviously make this change useless if we didn't want to set `WriteConnectionSecretToReference` for a resource.

Reply:
That's exactly what I had in mind, too. Thanks!